portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/fileinstall/FileInstall.testcase

```diff
@@ -121,6 +154,171 @@ definition {
 			configurationScope = "System Scope");
 	}
 
+	@description = "Verify an existing factory config can be modified."
+	@priority = "3"
+	test ModifyFactoryConfig {
+		property osgi.module.configuration.file.names = "com.liferay.portal.security.auth.verifier.internal.tunnel.configuration.TunnelAuthVerifierConfiguration~test.config";
+		property osgi.module.configurations = "enabled=\"true\"${line.separator}hostsAllowed=\"127.0.0.1,SERVER_IP\"${line.separator}serviceAccessPolicyName=\"SYSTEM_USER_PASSWORD\"${line.separator}urlsIncludes=\"/api/liferay/original\"";
+		property test.name.skip.portal.instance = "FileInstall#ModifyFactoryConfig";
+
+		task ("When a factory configuration is modified") {
+			OSGiConfig.deployOSGiConfigFile(
+				osgiConfigFileName = "com.liferay.portal.security.auth.verifier.internal.tunnel.configuration.TunnelAuthVerifierConfiguration~test.config",
+				osgiConfigs = "enabled=\"false\",hostsAllowed=\"127.0.0.1\",serviceAccessPolicyName=\"SYSTEM_USER_PASSWORD_2\",urlsIncludes=\"/api/liferay/new\"");
+		}
+
+		task ("Then the factory configuration should be successfully updated") {
+			Portlet.shutdownServer();
```

Comment:
This is a bit excessive, since normally OSGi configurations take around 5-10 seconds to update depending on the polling rate, shutting down and starting portal back up can be a bit heavy when used un-necessarily. The most ideal way would be to wait for the Configuration to output a log that it is done updating, but since that is not yet supported, I am okay with this for now, just wanted to let you know how this works.